src/api/nutrition.ts

```diff
@@ -1,54 +1,82 @@
 import axios, { AxiosRequestConfig } from 'axios';
 import localForage from 'localforage';
+import moment from 'moment';
+import cloneDeep from 'lodash/cloneDeep';
+
 import { showLoader } from '@/shared/decorators';
 
 import { appKey, appID } from '../../credentials';
-import { IDailyIntake, ISearchItem, IHeaders } from '../models/nutrition';
+
+import { IDailyIntake, IHeaders, FoodInfo, Food, Meal } from '@/models/nutrition';
+import { FoodListItem } from '@/components/modules/nutrition/add-food/food-list-item.model';
 
 import { dailyIntake } from '@/mocks/daily-intake';
 
 enum API_ENDPOINTS {
-    SEARCH = 'https://api.nutritionix.com/v2/search',
-    ITEM = 'https://api.nutritionix.com/v2/item'
+  SEARCH = 'https://api.nutritionix.com/v2/search',
+  ITEM = 'https://api.nutritionix.com/v2/item'
 }
 
 export class NutritionAPI {
-    private static localForageStore: LocalForage;
-
-    constructor() {
-        NutritionAPI.localForageStore = localForage.createInstance({name: 'nutritionDatabase'});
-        NutritionAPI.localForageStore.setItem('2019-08-13', dailyIntake('2019-08-13'));
-        NutritionAPI.localForageStore.setItem('2019-08-14', dailyIntake('2019-08-14'));
-    }
-
-    private get headers(): IHeaders {
-        return {
-            'x-app-id': appID,
-            'x-app-key': appKey,
-        };
-    }
-
-    @showLoader
-    public async getSearchItems(query: string): Promise<ISearchItem[]> {
-        const config: AxiosRequestConfig = {
-            params: {
-                q: query,
-            },
-            headers: this.headers,
-        };
-        return axios.get(API_ENDPOINTS.SEARCH, config).then((result: any) => result.data.results);
-    }
-
-    @showLoader
-    public async getItem(id: string): Promise<any> {
-        const config: AxiosRequestConfig = {
-            headers: this.headers,
-        };
-        return axios.get(`${API_ENDPOINTS.ITEM}/${id}`, config);
-    }
-
-    @showLoader
-    public async getMock(day: string): Promise<IDailyIntake> {
-        return NutritionAPI.localForageStore.getItem(day);
-    }
+  private static localForageStore: LocalForage;
+
+  constructor() {
+    NutritionAPI.localForageStore = localForage.createInstance({
+      name: 'nutritionDatabase',
+    });
+    NutritionAPI.localForageStore.setItem('2019-09-09', dailyIntake('2019-09-09'));
+  }
+
+  private get headers(): IHeaders {
+    return {
+      'x-app-id': appID,
+      'x-app-key': appKey,
+    };
+  }
+
+  @showLoader
+  public async getSearchItems(query: string): Promise<FoodListItem[]> {
+    const config: AxiosRequestConfig = {
+      params: {
+        q: query,
+        offset: 40,
+        limit: 40,
+      },
+      headers: this.headers,
+    };
+    return axios
+      .get(API_ENDPOINTS.SEARCH, config)
+      .then((result: any) => result.data.results);
```

Comment:
Would it be possible to use some more specific type here? Also I see `any` in some other places, could those be changed as well?

Reply:
As I'm using third party API I purposely don't add typing here. I'm considering adding just partial typing like `result: {data: Type}`